src/mqtt-client/client.js

```diff
@@ -25,93 +26,59 @@ const client = mqtt.connect({
     auth: `${config.mqtt.username}:${config.mqtt.password}`
 });
 
-let blockPublishing = false,
-    publishingQueue = new Set();
+const deviceInfoPublishQueue = uniqueQueue();
+const deviceInfoPublishQueueLock = lock();
+let deviceInfoPublishQueueClearTimeout = 0;
 
-client.on('connect', onConnect);
+client
+    .on('connect', onMqttConnect)
+    .on('message', onMqttMessage);
 
-function onConnect() {
-    client.subscribe(`${MQTT_INPUT_TOPIC_PREFIX}#`, onSubscribed);
-
-    output.stream.subscribe(event => {
-        let message, topic = MQTT_OUTPUT_TOPIC_PREFIX;
-
-        if ( event.event === DEVICE_STATUS_EVENT) {
-            message = event.value;
-            topic += event.device;
-        } else  if (event.event === DEVICE_INFO_EVENT) {
-
-            publishDeviceInfoEvent(event.device);
-            return;
-
-        } else {
-            debug(`Unknown output stream event received: '${JSON.stringify(event)}'`);
-            return;
-        }
-
-        client.publish(topic, message);
-    });
+function onMqttConnect() {
+    client.subscribe(`${INPUT}#`);
+    output.stream.subscribe(onInputEvent);
 }
 
-function onSubscribed() {
-    client.on('message', function (topic, rawMessage) {
-        debug(`MQTT >>. Got message: topic '${topic}', message: '${rawMessage.toString()}'`);
-        let message = '';
-        try {
-            message = JSON.parse(rawMessage);
-        } catch (e) {
-            message = rawMessage.toString();
-        }
-        let isDeviceInfo = typeof message === 'object';
-        let event,
-            deviceName = topic.split('/').pop();
-
-        if (isDeviceInfo) {
-
-            publishingQueue.delete(deviceName);
-            blockPublishing = false;
-            publishDeviceInfoEvent();
-
-            event  = {
-                event: 'device-info',
-                device: deviceName,
-                value: message
-            };
-        }
-        else {
-            event = {
-                event: 'status',
-                device: deviceName,
-                value: rawMessage.toString()
-            };
-        }
-        input.write(event);
-        debug(`Added to input stream event: '${JSON.stringify(event)}'`);
-    });
+function onMqttMessage(topic, rawMessage) {
+    debug(`MQTT >>. Got message: topic '${topic}', message: '${rawMessage.toString()}'`);
+    const event = convertMqttMessageToEvent(topic, rawMessage);
+    if (event.event === DEVICE_INFO) {
+        deviceInfoPublishQueue.remove(event.device);
+        deviceInfoPublishQueueLock.unlock();
+        publishDeviceInfoEvent();
+    }
+    input.write(event);
+    debug(`Added to input stream event: '${JSON.stringify(event)}'`);
 }
 
-function publishDeviceInfoEvent (device) {
-    let timeoutTimer;
-
-    if (device) {
-        publishingQueue.add(device);
+function onInputEvent(event) {
```

Comment:
as you can see bottom there is not only event property is used,  
for instance: JSON.stringify(event) !fix